awx/main/tasks.py

```diff
@@ -1001,11 +1002,17 @@ def should_use_proot(self, instance):
         return False
 
     def build_inventory(self, instance, private_data_dir):
-        script_params = dict(hostvars=True)
+        script_params = dict(hostvars=True, towervars=True)
         if hasattr(instance, 'job_slice_number'):
             script_params['slice_number'] = instance.job_slice_number
             script_params['slice_count'] = instance.job_slice_count
         script_data = instance.inventory.get_script_data(**script_params)
+        # maintain a list of host_name --> host_id
+        # so we can associate emitted events to Host objects
+        self.host_map = {
+            hostname: hv['remote_tower_id']
+            for hostname, hv in script_data.get('_meta', {}).get('hostvars', {}).items()
```

Comment:
using `towervars=True` will add more user-facing extra_vars, which I think needs a documentation note if we go with this.

But I see that it's not truly necessary given your implementation here. Some alternatives to adding new playbook variables:

 - pop off the var, `hv.pop('remote_tower_id')`
 - restructure the code so that `get_script_data` produces a `host_map` directly as another returned item

We've been adding a lot of playbook vars, and adding this is no big deal. The precedent exists, but the precedent is also that they are documented.

https://docs.ansible.com/ansible-tower/3.5.0/html/userguide/job_templates.html#launch-a-job-template

All of the options here are at least a _little_ painful, but fairly minor pain.

Reply:
Decided to pop this variable off the dict in `tasks.py`.